spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -131,12 +133,12 @@ public SparkTable(Table icebergTable, boolean refreshEagerly) {
   public SparkTable(Table icebergTable, String branch, boolean refreshEagerly) {
     this(icebergTable, refreshEagerly);
     this.branch = branch;
-    ValidationException.check(
```

Comment:
I don't really understand these changes.
I think everything should be reverted except changes in `hashCode` and `equals`.